app/models/user.rb

```diff
@@ -48,6 +48,21 @@ def email_changed?
     !persisted? || (emails.loaded? && emails.each.any?(&:changed))
   end
 
+  # Unset current primary email
+  def unset_primary_email
+    primary_email = emails.find(&:primary?)
+    primary_email.update_attributes!(primary: false) if primary_email
+  end
+
+  # Pick first email and set it as primary email.
+  #
+  # @raise [ActiveRecord::Rollback] Raises if cannot find any email.
+  def set_next_email_as_primary
+    next_email = emails.first
+    fail ActiveRecord::Rollback unless next_email
```

Comment:
This is dangerous because what if there's no transaction? Who calls this method?